Could not load file or assembly Microsoft.DotNet.PlatformAbstractions under `dotnet tye`

I see the following in logs for an azure-pipeline job for a dotnet core 3.1 application. These lines appear in the [tye](https://github.com/dotnet/tye) step of [this pipeline](https://github.com/staff0rd/DigitalIcebreakers/blob/a0205b56334a199606766d8a13eda392627ec727/azure-pipelines.yml#L57-L58).

**Regardless of this error, the versions still seem to be updated correctly**, for both the pipeline, dotnet assembly and node script. As such I post this here for informative purposes only.

```
2020-04-13T05:26:25.9406642Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly.
2020-04-13T05:26:25.9408785Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception.
2020-04-13T05:26:25.9410785Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:  ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.DotNet.PlatformAbstractions, Version=2.1.0.0, Culture=neutral, PublicKey***=adb9793829ddae60'. The system cannot find the file specified.
2020-04-13T05:26:25.9412046Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018: 
2020-04-13T05:26:25.9413899Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018: File name: 'Microsoft.DotNet.PlatformAbstractions, Version=2.1.0.0, Culture=neutral, PublicKey***=adb9793829ddae60'
2020-04-13T05:26:25.9415112Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.GitLoaderContext.GetNativeLibraryDirectory()
2020-04-13T05:26:25.9416177Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.GitLoaderContext.LoadUnmanagedDll(String unmanagedDllName)
2020-04-13T05:26:25.9417326Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext)
2020-04-13T05:26:25.9418454Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init()
2020-04-13T05:26:25.9419623Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.InitializeNativeLibrary()
2020-04-13T05:26:25.9420629Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor()
2020-04-13T05:26:25.9421673Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018: 
2020-04-13T05:26:25.9422629Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018: 
2020-04-13T05:26:25.9423869Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    --- End of inner exception stack trace ---
2020-04-13T05:26:25.9424971Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_opts(Int32 option, UInt32 level, String path)
2020-04-13T05:26:25.9426094Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.git_libgit2_opts_set_search_path(ConfigurationLevel level, String path)
2020-04-13T05:26:25.9427226Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.GlobalSettings.SetConfigSearchPaths(ConfigurationLevel level, String[] paths)
2020-04-13T05:26:25.9428371Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.GitExtensions.OpenGitRepo(String pathUnderGitRepo, Boolean useDefaultConfigSearchPaths)
2020-04-13T05:26:25.9429732Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot)
2020-04-13T05:26:25.9430984Z /home/vsts/.nuget/packages/nerdbank.gitversioning/3.1.74/build/Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner()
2020-04-13T05:26:25.9431677Z     0 Warning(s)
2020-04-13T05:26:25.9432033Z     1 Error(s)
```

This just broke me locally on Windows

```
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:  ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.DotNet.PlatformAbstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified. [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018: File name: 'Microsoft.DotNet.PlatformAbstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.GitLoaderContext.GetNativeLibraryDirectory() [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.GitLoaderContext.LoadUnmanagedDll(String unmanagedDllName) [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.InitializeNativeLibrary() [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    --- End of inner exception stack trace --- [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_opts(Int32 option, UInt32 level, String path) [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.git_libgit2_opts_set_search_path(ConfigurationLevel level, String path) [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at LibGit2Sharp.GlobalSettings.SetConfigSearchPaths(ConfigurationLevel level, String[] paths) [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.GitExtensions.OpenGitRepo(String pathUnderGitRepo, Boolean useDefaultConfigSearchPaths) [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot) [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
C:\Users\Claire\.nuget\packages\nerdbank.gitversioning\3.1.91\build\Nerdbank.GitVersioning.targets(71,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner() [D:\dev\NuGetPackageExplorer\Types\Types.csproj]
    0 Warning(s)
    1 Error(s)
```

--- a/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.Tasks.csproj
+++ b/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.Tasks.csproj
@@ -33,6 +33,7 @@
       <TfmSpecificPackageFile Include="
                         $(OutputPath)LibGit2Sharp.dll*;
                         $(OutputPath)MSBuildExtensionTask.dll;
+                        $(OutputPath)Microsoft.DotNet.PlatformAbstractions.dll;
                         $(OutputPath)Nerdbank.GitVersioning.*dll;
                         $(OutputPath)Newtonsoft.Json.dll;
                         $(OutputPath)Validation.dll;

--- a/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.nuspec
+++ b/src/Nerdbank.GitVersioning.Tasks/Nerdbank.GitVersioning.nuspec
@@ -19,7 +19,8 @@ IMPORTANT: The 3.x release may produce a different version height than prior maj
   <files>
     <file src="$BaseOutputPath$net461\LibGit2Sharp.dll" target="build\MSBuildFull\LibGit2Sharp.dll" />
     <file src="$BaseOutputPath$net461\LibGit2Sharp.dll.config" target="build\MSBuildFull\LibGit2Sharp.dll.config" />
-    <file src="$BaseOutputPath$net461\NerdBank.GitVersioning.dll" target="build\MSBuildFull\NerdBank.GitVersioning.dll" />
+    <file src="$BaseOutputPath$net461\Microsoft.DotNet.PlatformAbstractions.dll" target="build\MSBuildFull\Microsoft.DotNet.PlatformAbstractions.dll" />
+    <file src="$BaseOutputPath$net461\NerdBank.GitVersioning.dll" target="build\MSBuildFull\NerdBank.GitVersioning.dll" />    
     <file src="$BaseOutputPath$net461\Nerdbank.GitVersioning.Tasks.dll" target="build\MSBuildFull\Nerdbank.GitVersioning.Tasks.dll" />
     <file src="$BaseOutputPath$net461\Newtonsoft.Json.dll" target="build\MSBuildFull\Newtonsoft.Json.dll" />
     <file src="$BaseOutputPath$net461\Validation.dll" target="build\MSBuildFull\Validation.dll" />
@@ -35,6 +36,7 @@ IMPORTANT: The 3.x release may produce a different version height than prior maj
 
     <file src="$LibGit2SharpNativeBinaries$libgit2\LibGit2Sharp.dll.config" target="build\MSBuildCore\LibGit2Sharp.dll.config" />
     <file src="$BaseOutputPath$netcoreapp2.0\LibGit2Sharp.dll" target="build\MSBuildCore\LibGit2Sharp.dll" />
+    <file src="$BaseOutputPath$netcoreapp2.0\Microsoft.DotNet.PlatformAbstractions.dll" target="build\MSBuildCore\Microsoft.DotNet.PlatformAbstractions.dll" />
     <file src="$BaseOutputPath$netcoreapp2.0\NerdBank.GitVersioning.dll" target="build\MSBuildCore\NerdBank.GitVersioning.dll" />
     <file src="$BaseOutputPath$netcoreapp2.0\Nerdbank.GitVersioning.Tasks.dll" target="build\MSBuildCore\Nerdbank.GitVersioning.Tasks.dll" />
     <file src="$BaseOutputPath$netcoreapp2.0\Newtonsoft.Json.dll" target="build\MSBuildCore\Newtonsoft.Json.dll" />

